Mathlib/Data/Finsupp/Multiset.lean

```diff
@@ -251,3 +256,46 @@ end Finsupp
 theorem Multiset.toFinsupp_strictMono [DecidableEq ι] : StrictMono (@Multiset.toFinsupp ι _) :=
   (@Finsupp.orderIsoMultiset ι).symm.strictMono
 #align multiset.to_finsupp_strict_mono Multiset.toFinsupp_strictMono
+
+namespace Sym
+
+variable (α)
+variable [DecidableEq α] (n : ℕ)
+
+/-- The `n`th symmetric power of a type `α` is naturally equivalent to the subtype of
+finitely-supported maps `α →₀ ℕ` with total mass `n`.
+
+See also `Sym.equivNatSumOfFintype` when `α` is finite. -/
+def equivNatSum :
+    Sym α n ≃ {P : α →₀ ℕ // P.sum (fun _ ↦ id) = n} :=
+  (Multiset.toFinsupp.toEquiv.image _).trans <| Equiv.Set.ofEq <| by
```

Comment:
Is `Equiv.subtypeEquiv` any more direct here?

Reply:
Yes, `(Multiset.toFinsupp.toEquiv).subtypeEquiv fun a => by simp`!

Reply:
Ooooh, nice. Thanks!